src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs

```diff
@@ -280,47 +284,82 @@ public async Task When_Text_Changed_SuggestionChosen()
 			Assert.AreEqual(AutoSuggestionBoxTextChangeReason.SuggestionChosen, reason);
 		}
 
-
 		[TestMethod]
-		public async Task When_Text_Changed_Sequence()
+		[DataRow(true)]
+		[DataRow(false)]
+		public async Task When_Text_Changed_Sequence(bool waitBetweenActions)
 		{
-			var SUT = new AutoSuggestBox();
-			SUT.ItemsSource = new List<string>() { "ab", "abc", "abcde" };
+			var SUT = new AutoSuggestBox()
+			{
+				ItemsSource = new List<string>() { "ab", "abc", "abcde" }
+			};
 			WindowHelper.WindowContent = SUT;
 			await WindowHelper.WaitForIdle();
-			bool eventRaised = false;
+
+			var textBox = (TextBox)SUT.GetTemplateChild("TextBox");
+
+			var expectations = new List<AutoSuggestionBoxTextChangeReason>();
 			var reasons = new List<AutoSuggestionBoxTextChangeReason>();
-			var counter = 0;
 			SUT.TextChanged += (s, e) =>
 			{
 				reasons.Add(e.Reason);
-				if (++counter == 7)
-				{
-					eventRaised = true;
-				}
 			};
-			var textBox = (TextBox)SUT.GetTemplateChild("TextBox");
+
+			expectations.Add(SuggestionChosen);
 			SUT.Focus(FocusState.Programmatic);
 			SUT.ChoseItem("ab");
+			await Wait();
+
+			expectations.Add(ProgrammaticChange);
 			SUT.Text = "other";
+			await Wait();
+
+			expectations.Add(UserInput);
+			SUT.Focus(FocusState.Programmatic);
 			textBox.ProcessTextInput("manual");
+			await Wait();
+
+			expectations.Add(SuggestionChosen);
 			SUT.ChoseItem("ab");
-			textBox.ProcessTextInput("manual");
+			await Wait();
+
+			expectations.Add(UserInput);
+			SUT.Focus(FocusState.Programmatic);
+			textBox.ProcessTextinput("manual");
+			await Wait();
+
+			expectations.Add(ProgrammaticChange);
+			SUT.Focus(FocusState.Programmatic);
 			SUT.Text = "other";
+			await Wait();
+
+			expectations.Add(SuggestionChosen);
 			SUT.ChoseItem("ab");
+			await Wait();
 
-			await WindowHelper.WaitFor(() => eventRaised);
-			CollectionAssert.AreEquivalent(
-				new[] {
-					AutoSuggestionBoxTextChangeReason.SuggestionChosen,
-					AutoSuggestionBoxTextChangeReason.ProgrammaticChange,
-					AutoSuggestionBoxTextChangeReason.UserInput,
-					AutoSuggestionBoxTextChangeReason.SuggestionChosen,
-					AutoSuggestionBoxTextChangeReason.UserInput,
-					AutoSuggestionBoxTextChangeReason.ProgrammaticChange,
-					AutoSuggestionBoxTextChangeReason.SuggestionChosen
-				},
-				reasons);
+			await WindowHelper.WaitForIdle();
+
+#if __SKIA__
+			// skia is closer to what happens on WinUI. On WinUI, if there is no delay between changes, 
+			// AutoSuggestBox.TextChanged is fired once (but TextBox.TextChanged fires everytime)
```

Comment:
@ramezgerges 
is this even intended? i assuming all of uno code would behave similar, unless underlying platform or platform-specific code suggest otherwise
this feel like a bug that should be fixed, or [Ignore]'d rather than getting a special passing setup...

Reply:
The intended behaviour is to match WinUI (I don't think the WinUI devs would classify this as a bug). The behviour in Uno is not platform-agnostic due to the way TextBox works, which differs between the managed (skia) and native (mobile) implementations. I don't remember very well, but I think AutoSuggestBox.TextChanges has a real reason for behaving like this. Conceptually, I think AutoSuggestBox only cares about entire "text runs", not individual character changes. This reduces the amount of suggestion updates that need to happen (e.g. like how server-side input-validation doesn't happen on each keystroke).